sdk/src/main/java/io/horizen/sc2sc/CrossChainMessageValidator.java

```diff
@@ -0,0 +1,43 @@
+package io.horizen.sc2sc;
+
+import io.horizen.account.sc2sc.AccountCrossChainMessage;
+import io.horizen.utils.Constants;
+
+public final class CrossChainMessageValidator {
+    private final static String SENDER_SIDECHAIN_ID_ERROR_MESSAGE = "Sender sidechain id must be 32 bytes long";
+    private final static String RECEIVER_SIDECHAIN_ID_ERROR_MESSAGE = "Receiver sidechain id must be 32 bytes long";
+    private final static String SENDER_ADDRESS_ERROR_MESSAGE = "Sender address length is not correct";
+    private final static String RECEIVER_ADDRESS_ERROR_MESSAGE = "Receiver address length is not correct";
+    public void validateMessage(CrossChainMessage msg) {
+        validateMsgType(msg.getMessageType());
+        validateSidechainId(msg.getSenderSidechain(), SENDER_SIDECHAIN_ID_ERROR_MESSAGE);
+        validateSidechainId(msg.getReceiverSidechain(), RECEIVER_SIDECHAIN_ID_ERROR_MESSAGE);
+        validateAddress(msg.getSender(), SENDER_ADDRESS_ERROR_MESSAGE);
+        validateAddress(msg.getReceiver(), RECEIVER_ADDRESS_ERROR_MESSAGE);
+    }
+
+    public void validateMessage(AccountCrossChainMessage accMsg) {
+        validateMsgType(accMsg.messageType());
+        validateSidechainId(accMsg.receiverSidechain(), RECEIVER_SIDECHAIN_ID_ERROR_MESSAGE);
+        validateAddress(accMsg.sender(), SENDER_ADDRESS_ERROR_MESSAGE);
+        validateAddress(accMsg.receiver(), RECEIVER_ADDRESS_ERROR_MESSAGE);
+    }
+
+    private void validateMsgType(int msgType) {
+        if (msgType < 0) {
+            throw new IllegalArgumentException("CrossChain message type cannot be negative");
+        }
+    }
+
+    private void validateSidechainId(byte[] sidechainId, String exceptionMsg) {
+        if (sidechainId.length != Constants.SIDECHAIN_ID_SIZE()) {
+            throw new IllegalArgumentException(exceptionMsg);
+        }
+    }
+
+    private void validateAddress(byte[] address, String exceptionMsg) {
+        if (address.length != Constants.SIDECHAIN_ADDRESS_SIZE() && address.length != Constants.ABI_ADDRESS_SIZE()) {
```

Comment:
These limits comes from the the type of propositions currently supported in our sidechains, but in the future could be more.
To make it more clear I would remove from Constants ABI_ADDRESS_SIZE  and SIDECHAIN_ADDRESS_SIZE and use in the checks these constants already defined:
io.horizen.account.proposition.AddressProposition.LENGTH
io.horizen.proposition;.PublicKey25519Proposition.KEY_LENGTH